Realm/RLMArray.mm

```diff
@@ -437,3 +438,23 @@ - (instancetype)reversedSortDescriptor {
 }
 
 @end
+
+@interface RLMArray (Handover) <RLMThreadConfined_Private>
+@end
+
+@implementation RLMArray (Handover)
+
+- (realm::AnyThreadConfined)rlm_handoverData {
+    @throw RLMException(@"Unexpected handover of unmanaged `RLMArray`");
```

Comment:
Well, I would like to call a no-return function here. Any suggestions if not throwing an exception? @tgoyne